roles/mirror_ocp_release/tasks/installer.yml

```diff
@@ -10,12 +10,19 @@
 - name: "Extract installer command from release image"
   when:
     - mor_force | bool or not target.stat.exists
-  ansible.builtin.command: >
-    {{ mor_oc }} adm release extract
-    --registry-config={{ mor_auths_file }}
-    --command={{ mor_installer }}
-    --from {{ mor_pull_url }}
-    --to "{{ mor_cache_dir }}/{{ mor_version }}"
+  ansible.builtin.shell: |
+    flock -x -n -E 200 "{{ mor_cache_dir }}/{{ mor_version }}/installer.lock" -c '
+    {{ mor_oc }} adm release extract \
+    --registry-config="{{ mor_auths_file }}" \
+    --command="{{ mor_installer }}" \
+    --from "{{ mor_pull_url }}" \
+    --to "{{ mor_cache_dir }}/{{ mor_version }}"'
+
+    if [ "$?" -eq 200 ]
+    then
+      echo "Waiting for lock release"
+      flock -x -w 30 "{{ mor_cache_dir }}/{{ mor_version }}/installer.lock" -c 'echo "Lock release"'
+    fi
```

Comment:
The new logic introduces a problem with the retries, these are never done because the result is always successful. When there is a failure in the task inside flock, e.g. a timeout, the task is not retried.

Reply:
Good point!
Let's see if this new version is better.